### ISSUE TITLE
Allow gene-based grouping for dittoPlot

### ISSUE THREAD

**Issue by j-andrews7** [dtm2451/dittoSeq#99]
It is sometimes useful to group samples by gene along the x-axis, like so:

![image](https://user-images.githubusercontent.com/10225716/139952870-32cd9265-2b8c-43b9-9bfa-b29691d11776.png)

For small numbers of genes, this can provide a more compact figure than faceting as `multi_dittoPlot` does.

**Comment by dtm2451** [comment]
Hmmm, do A-G represent genes, or sets of genes?, here and treatment is a sample grouping?

**Comment by j-andrews7** [comment]
A-G would be genes. The treatment would be a sample grouping, sorry, I could have clarified better in the OP.

**Comment by dtm2451** [comment]
No worries!  Just wanted to clarify =).  I like the idea and will ponder on how best to implement.  Potentially could be an alternate plotting functionality in the back-end, but accessible with just `multi_dittoPlot(..., group.by = "vars")`.

**Comment by j-andrews7** [comment]
Ohh, very nice, thanks for the addition!

**Comment by dtm2451** [comment]
Yep!  You can't do it in the "release" version yet, but you can if you install from github.

It'll be available from dittoSeq 1.10 after the next Bioconductor release in November.

Just give the desired set of `var`s to `var` (any metadata or genes of the object).
Customization Details: You can use `multivar.aes` to switch between displaying those vars in faceting/splitting, color, or x-axis groupings.  And when using splitting, and `split.by = "<something_else>"`, you can use `multivar.split.dir` to control which direction the `var` data goes .

Install with:
```
if (!requireNamespace("remotes")) { install.packages("remotes") }
remotes::install_github("dtm2451/dittoSeq")
```

Usage = any of these:
```
library(dittoSeq)
example("importDittoBulk", echo = FALSE)
dittoPlot(object = myRNA, var = c("gene1", "gene2", "gene3"), group.by = "timepoint") # default: `multivar.aes = "split"`
dittoPlot(object = myRNA, var = c("gene1", "gene2", "gene3"), group.by = "timepoint", multivar.aes = "group")
dittoPlot(object = myRNA, var = c("gene1", "gene2", "gene3"), group.by = "timepoint", multivar.aes = "color")
```

![image](https://user-images.githubusercontent.com/38870347/176735145-a0965bdf-730c-445d-9a73-e15fbb823223.png)
![image](https://user-images.githubusercontent.com/38870347/176735380-873f56a1-9fed-4a62-b431-435f921376d8.png)
![image](https://user-images.githubusercontent.com/38870347/176735487-656ab0f3-7fd8-43ba-bae8-77b2b7840306.png)

```
# Or with extra split.by variable:
dittoPlot(object = myRNA, var = c("gene1", "gene2", "gene3"), group.by = "timepoint",
    split.by = "clustering") # default: `multivar.split.dir = "col"`
dittoPlot(object = myRNA, var = c("gene1", "gene2", "gene3"), group.by = "timepoint",
    split.by = "clustering", multivar.split.dir = "row")
```

![image](https://user-images.githubusercontent.com/38870347/194438842-afed09e8-89d9-442b-9c4d-0f554d87e006.png)
![image](https://user-images.githubusercontent.com/38870347/194438873-0c9f2394-c0c5-4a9e-b4cd-7b2280982291.png)

**Comment by dtm2451** [comment]
> Is there a way to produce the dittobarplot similar to the figure above, i.e. the group-by is plotted within one axis and not split between different graphs?

Oops, on re-read, noticing you asked about ditto***Bar***Plot().  Can you clarify how you mean, @vivianschu?

Stab#2... dittoFreqPlot() splits the var-levels / cell types of a dittoBarPlot() up, so if you mean like a dittoFreqPlot() with the cell types all side-by-side: I hadn't thought to build `multivar.aes`-like control for that function, but you could achieve a similar feel by giving `split.nrow = 1`...
```
dittoFreqPlot(myRNA, "clustering", group.by = "groups", sample.by = "sample",
    split.nrow = 1)
```
![image](https://user-images.githubusercontent.com/38870347/194576848-8e187dc4-383f-4a1e-918c-60b09fc3f866.png)

**Comment by vivianschu** [comment]
My bad! I meant dittoBoxPlot, not dittoBarPlot.

I tested out your suggestions on how to produce those plots (particularly interested in how to reproduce the last one), however they generate the same result with box plots stacked instead of in one row.

<img width="853" alt="image" src="https://user-images.githubusercontent.com/33632082/194615782-e5fa5202-4188-445c-92b1-dec2c518f13d.png">

I am trying to illustrate a gene's expression level among each cluster, comparing the control samples to the treatment.

Here is my pseudoscript below, which produce the image above:
```
if (!requireNamespace("remotes")) { install.packages("remotes") }
remotes::install_github("dtm2451/dittoSeq")

library(dittoSeq)

dittoBoxPlot(seurat_object, "Gene", group.by = "condition", split.by = "cluster", multivar.split.dir = "row") 
```

**Comment by dtm2451** [comment]
Ohhhh!  You don't need the new version unless you wanted to do something like:
```
dittoBoxPlot(seurat_object, c("Gene1", "Gene2"), ...)
```

This `issue` was about plotting results for multiple genes / vars in a single plot.

If you are just trying to put all your `split.by = "cluster"` facets in a single row, all you need to do is to add `split.nrow = 1`!

(And fyi, syntax for dittoBoxPlot is exactly the same as for dittoPlot. So as you may already have noticed, you can modify any example code that uses dittoPlot by just adding in the "Box".  The ONLY difference between the two lies in their default values for the 'plots' input.)

**Comment by vivianschu** [comment]
Wonderful!! Thank you so much - I didn't realize the solution was so simple.

Last question -- is there a method to place the group.by labels on the bottom instead of the top?
<img width="965" alt="image" src="https://user-images.githubusercontent.com/33632082/194630880-6083d115-cbce-4861-bc6e-258e5a83bb92.png">

**Comment by dtm2451** [comment]
Yup!  The documentation path is a bit hard to follow, but I tried to describe in the docs for the 'split.adjust' input how you can use it to pass tweaks into the faceting functions.  Since you're only giving one variable to 'split.by', you'd follow `?facet_wrap` for options.  And looking there, I see a 'strip.position' input.  Try adding: `split.adjust = list(strip.position = "bottom")`

**Comment by vivianschu** [comment]
Thanks so much!!
